app/src/main/kotlin/com/wire/android/ui/login/Login.kt

```diff
@@ -0,0 +1,204 @@
+    )
+}
+
+@Composable
+private fun ForgotPasswordLabel(modifier: Modifier) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = modifier) {
+        val context = LocalContext.current
+        Text(
+            text = stringResource(R.string.login_forgot_password),
+            style = MaterialTheme.typography.body1.copy(
+                textDecoration = TextDecoration.Underline,
+                color = MaterialTheme.colors.primary
+            ),
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = null,
+                    onClick = { Toast.makeText(context, "Forgot password click 💥", Toast.LENGTH_SHORT).show() } //TODO
+                )
+        )
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun LoginButton(modifier: Modifier, email: String, password: String) {
+    var isLoading by remember { mutableStateOf(false) }
+    val interactionSource = remember { MutableInteractionSource() }
+    Column(modifier = modifier) {
+        val enabled = validInput(email, password) && !isLoading
+        Button(
+            interactionSource = interactionSource,
+            shape = RoundedCornerShape(16.dp),
+            elevation = ButtonDefaults.elevation(0.dp, 0.dp, 0.dp),
+            colors = AnimatedButtonColors(enabled = enabled),
+            onClick = { isLoading = true }, //TODO
+            enabled = validInput(email, password) && !isLoading,
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(48.dp)
+        ) {
+            Box(modifier = Modifier.fillMaxWidth()) {
+                val text = if(isLoading) stringResource(R.string.label_logging_in) else stringResource(R.string.label_login)
+                Text(
+                    text = text,
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.button,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 32.dp),
+                )
+                androidx.compose.animation.AnimatedVisibility(
+                    modifier = Modifier.align(Alignment.CenterEnd),
+                    visible = isLoading,
+                    enter =  fadeIn() + scaleIn(),
+                    exit =  fadeOut() + scaleOut()
+                ) {
+                        CircularProgressIndicator(
+                            modifier = Modifier.size(16.dp).align(Alignment.CenterEnd),
+                            color = LocalContentColor.current.copy(alpha = LocalContentAlpha.current),
+                            strokeWidth = 2.dp)
+                }
+            }
+        }
+    }
+}
+
+private fun validInput(email: String, password: String): Boolean =
+    email.isNotEmpty() && password.isNotEmpty()
```

Comment:
same here, it can be handeled by the viewModel

Reply:
Yes it can be in the ViewModel, current solution is temporary just to get the UI working, I assumed that this PR would only add UI, the logic and ViewModel would be added in the next